src/comm.cc

```diff
@@ -1666,10 +1667,12 @@ commHalfClosedCheck(void *)
         Comm::ConnectionPointer c = new Comm::Connection; // XXX: temporary. make HalfClosed a list of these.
         c->fd = *i;
         if (!fd_table[c->fd].halfClosedReader) { // not reading already
-            AsyncCall::Pointer call = commCbCall(5,4, "commHalfClosedReader",
-                                                 CommIoCbPtrFun(&commHalfClosedReader, NULL));
-            Comm::Read(c, call);
-            fd_table[c->fd].halfClosedReader = call;
+            CallBack(fd_table[c->fd].codeContext, [c] {
```

Comment:
Can we avoid refcounting overheads here and help the compiler to inline this lambda?

```suggestion
            CallBack(fd_table[c->fd].codeContext, [&c] {
```

Sorry I did not realize that `c` is a smart pointer when looking at this code earlier.

Reply:
Done.